File: backend/src/api/database/models/doctor.js
const sequelize = require('../connection');

const { Sequelize, DataTypes } = require('sequelize');

const Doctor = sequelize.define('Doctor', {

  doctor_id : {
    type : DataTypes.INTEGER,
    primaryKey: true,
    allowNull : false,
    autoIncrement : true
  },
  doctor_name: {
    type: DataTypes.STRING,
    unique: true,
    allowNull: false,
  },
  doctor_phone_number: {
    type: DataTypes.STRING,
    unique: true,
    allowNull: false,
  },
  doctor_password : {
    type : DataTypes.STRING,
    allowNull : false,
  },
 

}, 
{
  // Additional options
}
);


module.exports = Doctor;

File: frontend/lib/models/doctor.dart
class Doctor{
  Doctor();
  late int doctor_id;
  String  doctor_name = "";
  String doctor_phone_number = "";
  String doctor_password = "";

  clone(){
    return Doctor.fromJson(this.toJson());
  }

  Doctor.fromJson(Map<String, dynamic>json){

    doctor_id = json['doctor_id'];
    doctor_name = json['doctor_name'];
    doctor_phone_number = json['doctor_phone_number'];
    doctor_password = json['doctor_password'];

  }

  Map<String,dynamic> toJson(){
    Map<String,dynamic> doctorJson={
      "doctor_id" : doctor_id,
      "doctor_name" : doctor_name,
      "doctor_phone_number" : doctor_phone_number,
      "doctor_password" : doctor_password,
    };
    return doctorJson;
  }
}
